OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -534,5 +534,21 @@ public IEnumerable<Actor> ActorsInWorld()
 		{
 			return bins.SelectMany(bin => bin.Actors.Where(actor => actor.IsInWorld));
 		}
+
+		public IEnumerable<Actor> GetActorsOnRay(CPos start, CPos end)
+		{
+			foreach (var pos in Raycaster.Raycast(start, end))
+			{
+				var actors = GetUnitsAt(pos);
+
+				if (actors.Any())
```

Comment:
This check is unnecessary.  The `foreach` will behave correctly if `actors` is empty.